src/opera/template/topology.py

```diff
@@ -100,8 +100,10 @@ def join(self, params):
 
                     value_expression_name = param_dict_keys[0]
                     try:
+                        entity_name = self.find_node(
```

Comment:
This is a little confusing - we are actually dealing here not with entity name but entity itself 
```suggestion
                        entity = self.find_node(
```

Reply:
Yes, true. Here, we are actually getting an entity but I named it ambiguously, so I have changed it to just `entity`.